src/frontend/loading.rs

```diff
@@ -26,10 +27,12 @@ impl Component for LoadingView {
             set_valign: gtk::Align::Center,
             set_vexpand: true,
             set_orientation: gtk::Orientation::Vertical,
+            set_spacing: 10,
 
-            gtk::Spinner {
-                start: (),
-                set_size_request: (50, 50),
+            gtk::ProgressBar {
+                #[watch]
+                set_fraction: model.progress,
+                set_show_text: true,
             },
```

Comment:
I may have not been super clear, but the goal wasn't necessarily to remove the spinner.

While it is probably fine to do the way you did it, I was thinking about high-level steps shown with spinner and detailed steps below progress bar.

High level step would be "Creating farmer" and low-level would be creation and benchmarking of individual farms under that farmer.

Like title and subtitle if it makes sense.

Reply:
at first  i want to implement a spinner like this 
![image](https://github.com/subspace/space-acres/assets/41407352/0c831348-43e8-4b7d-b0ea-3976c764c9b8)
GTK does not have native support.

and i am not think its a better idea to use title and subtitle. Implementing a complex solution may not provide much greater help. I think using a progress bar is more suitable, implementing a spinner with a similar effect as shown above could work as well.  this will more friendly for both small farm and big farm. 

for a more accurate for each farm, maybe need move LoadingStep type to backend, or add a notify in backend/farmer.rs file.  but benchmarking need more deep change in farmer.

Reply:
That is too complex for this purpose, not worth it.

I think simply adding title with main step above progress bar and making text below progress bar describing smaller inner steps will be sufficient.

For benchmarking we will indeed have to modify a farmer since it doesn't expose that information at the moment.

Reply:
![image](https://github.com/subspace/space-acres/assets/41407352/5a01c7ee-006c-4302-a7db-a64b693a0b75)
is this your meaning.

Reply:
Yes, exactly!

Reply:
simple set consensus node 20%, all farm occupy 60%, others occupy 20%.  if need more accurate, maybe others nearly nothing, consensus node and all farm split 100% equaly (100%/(farmer_len+1).

 
![image](https://github.com/subspace/space-acres/assets/41407352/83d2b468-e1c7-41a5-b744-b79b83ffd162)